src/utils/SessionLock.ts

```diff
@@ -242,13 +245,25 @@ export async function getSessionLock(onNewInstance: () => Promise<void>): Promis
             };
         });
 
+        let sleepExpired = false;
         const sleepPromise = new Promise((resolve) => {
-            setTimeout(resolve, remaining, undefined);
+            setTimeout(() => {
+                sleepExpired = true;
+                resolve(undefined);
+            }, remaining);
         });
 
         window.addEventListener("storage", onStorageUpdate!);
         await Promise.race([sleepPromise, storageUpdatePromise]);
```

Comment:
This already tells you whether the sleep or the storage update won the race. 

```suggestion
        const winner = await Promise.race([sleepPromise, storageUpdatePromise]);
        ...
        if (!(winner instanceof StorageEvent)) {
        	prefixedLogger.info("Existing claim went stale: proceeding with startup");
            break;
        }
        ...
        
```